plugins/inputs/jolokia/jolokia.go

```diff
@@ -252,7 +252,14 @@ func (j *Jolokia) Gather(acc telegraf.Accumulator) error {
 							switch t2 := v.(type) {
 							case map[string]interface{}:
 								for k2, v2 := range t2 {
-									fields[measurement+"_"+k+"_"+k2] = v2
+									switch t3 := v2.(type) {
+									case map[string]interface{}:
+										for k3, v3 := range t3 {
+											fields[measurement+"_"+k+"_"+k2+"_"+k3] = v3
+										}
```

Comment:
Totally agree. I don't think there ever will be more than 3 layers of nesting, but in a long term I think a recursive call is definitely a better solution.

We also need to be careful with recursion in general, in case there is some kind of issues might happen and recursion tree becomes too long. Probably limiting depth should be included.